src/gherkin/index.js

```diff
@@ -12,12 +17,18 @@ class StepDef {
     if (!Object.values(Token).includes(token)) {
       throw new Error('Invalid token type')
     }
+    pattern = pattern.toString()
+    this.type =
+      pattern.startsWith('/^') && pattern.endsWith('$/') ? patternTypes.REGEX : patternTypes.REGULAR
```

Comment:
regex don't need to have `^` or `$` those are optional

Reply:
Yes, that's true. But we don't have a proper way to know if a pattern is a regular or regex one. In our case, we usually put that in our step definition so used this approach. Do you have any suggestions on how to identify if a given pattern is a regex?